cmd/auto-updater/main.go

```diff
@@ -110,11 +127,42 @@ func cmdRun(cmd *cobra.Command, args []string) (err error) {
 			}
 			log.Println("finished downloading", downloadURL, "to", tempArchive.Name())
 
-			// TODO next PRs
-			// * rename current binary into `storagenode.old.<release>.exe`.
-			// * unzip downloaded binary in place of current binary
-			// * compare extracted binary with version from suggested version
-			// * restart storage node service
+			extension := filepath.Ext(binaryLocation)
+			if extension != "" {
+				extension = "." + extension
+			}
+
+			dir, _ := filepath.Split(binaryLocation)
+			backupExec := filepath.Join(dir, "storagenode.old."+currentVersion.String()+extension)
+
+			if err = os.Rename(binaryLocation, backupExec); err != nil {
+				return err
+			}
+
+			err = unpackBinary(ctx, tempArchive.Name(), binaryLocation)
+			if err != nil {
+				return err
+			}
+
+			newVersion, err := binaryVersion(binaryLocation)
```

Comment:
How about renaming `newVersion` to `downloadedVersion`? It will be clearer when comparing later to `suggestedVersion`.

Reply:
good idea